src/metabase/warehouse_schema/models/table.clj

```diff
@@ -72,10 +101,22 @@
 
 (t2/define-before-update :model/Table
   [table]
-  (let [changes (t2/changes table)
+  (let [changes        (t2/changes table)
         original-table (t2/original table)
         current-active (:active original-table)
-        new-active (:active changes)]
+        new-active     (:active changes)]
+
+    ;; Prevent setting data_authority to :unknown (read-only value)
+    (when (= :unknown (some-> changes :data_authority keyword))
```

Comment:
how about we do this in transform?

Reply:
I feel like side effects in transforms are a last resort, as they're a bit surprising, and putting it in the hook feels more discoverable.

That said, I realize now there's nothing protecting us from code that initializes a table with this value though, so maybe you're right.

Will put it in the transform.